### PR TITLE
Fetching order data from the server

### PR DESCRIPTION
**Pull request by dousdogukan**
Fetching order data from the server and displaying them.
implement Loading and 404NotFound Pages,
add CTokenLabel component,
add Method class.

### CODE REVIEW

**Review by ruveyda-kisla — APPROVED** [review]

File: src/App.tsx [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
-  { value: 'stream', label: 'Stream' },
-  { value: 'single', label: 'Single' },
-  { value: 'vesting', label: 'Vesting' },
-];
-
-const networkTabs = [
-  { value: 'stellar', label: 'Stellar' },
-  { value: 'soroban', label: 'Soroban' },
-];
+const methodTabs = ['single', 'stream', 'vesting'];
```
Comment (ceydae) [inline src/App.tsx]: move to file in src/constant/methods.ts

File: src/App.tsx [hunk 2]
```diff
@@ -1,40 +1,45 @@
...
+  const [selectedNetwork, setSelectedNetwork] = useState<string>(
+    capitalizeFirstLetter(networkTabs[1]),
+  );
+  const [selectedMethod, setSelectedMethod] = useState<string>(
+    capitalizeFirstLetter(methodTabs[0]),
+  );
+
+  const { id } = useParams();
+
+  const { loading, loadingTime, data, error } = useGetOrderData(id);
```
Comment (ceydae) [inline src/App.tsx]: remove loadingTime

File: src/App.tsx [hunk 3]
```diff
@@ -45,7 +50,7 @@ export default function App() {
   const handleEmailChange = (e: ChangeEvent<HTMLInputElement>) => {
     const emailRegex = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
 
-    if (e.target.value && e.target.value.match(emailRegex)) {
+    if (e.target.value && emailRegex.test(e.target.value)) {
       setEmailAddress(e.target.value);
```
Comment (ceydae) [inline src/App.tsx]: add else condition for error message

File: src/App.tsx [hunk 4]
```diff
@@ -86,17 +102,24 @@ export default function App() {
...
           <p className="text-xs mt-[16px] w-3/4">
             Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor
             incididunt ut labore et dolore magna aliqua.
           </p>
+
           <div className="absolute gap-2 shadow-md rounded-[10px] py-2 pl-2 pr-3 inline-flex items-center z-10 -bottom-[30px] min-w-[240px] bg-white text-black">
-            <img src={hoodie} alt="shop" width={60} height={60} />
-            <p className="font-medium whitespace-nowrap">Amanda shop</p>
+            <img
+              src={data.user.logo}
```
Comment (ceydae) [inline src/App.tsx]: use default logo here

File: src/containers/Loading/index.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+
+import wagentLogo from '../../../public/images/wagentLogo.svg';
+
+interface LoadingProp {
+  loadingTime?: number;
+}
+
+const Loading = ({ loadingTime }: LoadingProp) => {
```
Comment (ceydae) [inline src/containers/Loading/index.tsx]: remove loadingTime, use a constant value of 2 seconds for loading

Comment (ceydae) [reply src/containers/Loading/index.tsx]: you can use animation for the progress bar

File: src/main.tsx [hunk 6]
```diff
@@ -1,11 +1,17 @@
...
 
 import App from './App.tsx';
 import './index.css';
 
 ReactDOM.createRoot(document.getElementById('root')!).render(
   <React.StrictMode>
-    <App />
+    <Router>
+      <Routes>
+        <Route path="/" element={<App />} />
```
Comment (ceydae) [inline src/main.tsx]: remove this line

File: src/utils/getOrderData.ts [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    let loadingTime = new Date().getTime();
+    const data = () => {
+      if (!id) {
+        return setOrderData({
+          loading: false,
+          loadingTime: 0,
+          data: null,
+          error: true,
+        });
+      }
```
Comment (ceydae) [inline src/utils/getOrderData.ts]: remove the code

**Review by ceydae — CHANGES_REQUESTED** [review]

**Review by ceydae — APPROVED** [review]

**Pull request merged**